Lit panics on initialization of USPV

This happens when you have specified an invalid node. Eventhough there's a fix for allowing to specify a hostname, the current master does not accept that. And if you only have invalid IPs like that, the `DialPeer` function will return a `nil` `net.Conn` but no error. This leads to a panic.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x584156]

goroutine 1 [running]:
github.com/mit-dci/lit/wire.NewMsgVersionFromConn(0x0, 0x0, 0x0, 0x0, 0x22, 0xc42034c690, 0x6)
	/usr/local/go/src/github.com/mit-dci/lit/wire/msgversion.go:249 +0x26
github.com/mit-dci/lit/uspv.(*SPVCon).Handshake(0xc42012e1e0, 0xc4203277f8, 0x1, 0x1, 0x0, 0x0)
	/usr/local/go/src/github.com/mit-dci/lit/uspv/init.go:128 +0x67
github.com/mit-dci/lit/uspv.(*SPVCon).Connect(0xc42012e1e0, 0xc4201fc084, 0x13, 0x0, 0x0)
	/usr/local/go/src/github.com/mit-dci/lit/uspv/init.go:223 +0x24a
github.com/mit-dci/lit/uspv.(*SPVCon).Start(0xc42012e1e0, 0x0, 0xc4201fc084, 0x13, 0xc4201fcd60, 0x12, 0x0, 0x0, 0x1049e20, 0xcb6e600d47652520, ...)
	/usr/local/go/src/github.com/mit-dci/lit/uspv/chainhook.go:123 +0x263
github.com/mit-dci/lit/wallit.NewWallit(0xc420512990, 0x0, 0xc4201fc084, 0x13, 0xc4201d2010, 0xa, 0x0, 0x0, 0x1049e20, 0x29e0ec3, ...)
	/usr/local/go/src/github.com/mit-dci/lit/wallit/init.go:68 +0x446
github.com/mit-dci/lit/qln.(*LitNode).LinkBaseWallet(0xc420216000, 0xc4201fc900, 0x0, 0xc4201fc084, 0x13, 0x0, 0x0, 0x1049e20, 0xa61fd0, 0x3)
	/usr/local/go/src/github.com/mit-dci/lit/qln/init.go:172 +0x17d
main.linkWallets(0xc420216000, 0xc4201fc900, 0xc420496600, 0xc4201fc108, 0x17)
	/usr/local/go/src/github.com/mit-dci/lit/lit.go:115 +0xa3d
main.main()
	/usr/local/go/src/github.com/mit-dci/lit/lit.go:285 +0x2e3
```

--- a/uspv/init.go
+++ b/uspv/init.go
@@ -118,6 +118,11 @@ func (s *SPVCon) DialNode(listOfNodes []string) (net.Conn, error) {
 		}
 		break
 	}
+
+	if con == nil {
+		return nil, fmt.Errorf("Failed to connect to a coin daemon")
+	}
+
 	return con, nil
 }
 

